src/engine/strat_engine/engine.rs

```diff
@@ -122,6 +154,39 @@ impl Engine for StratEngine {
     fn pools(&self) -> Vec<&Pool> {
         self.pools.into_iter().map(|x| x as &Pool).collect()
     }
+
+    fn get_eventable(&mut self) -> EngineResult<Option<Box<Eventable>>> {
+        Ok(Some(Box::new(DM::new()?)))
+    }
+
+    fn evented(&mut self) -> EngineResult<()> {
+        let device_list = DM::new()?.list_devices()?;
+
+        for &(ref dm_name, _, new_event_nr) in &device_list {
+            let new_event_nr = new_event_nr.expect("Supported DM versions always provide a value");
+            if let Some(prev_event_nr) = self.watched_dev_last_event_nrs.get(dm_name) {
+                if new_event_nr > *prev_event_nr {
+                    // Event! We know the dm name, but don't know which
+                    // pool
+                    for pool in &mut self.pools {
```

Comment:
Was it impossible to use ```find``` here?

Reply:
no, but it looks like this:
```
if let Some(pool) =
    (&mut self.pools)
        .into_iter()
        .find(|pool| pool.get_eventing_dev_names().contains(dm_name)) {
    pool.event_on(dm_name)?;
}
```

which doesn't seem much better than what's there now.